### PR TITLE
Live query java implementation

### PR DESCRIPTION
**Pull request by LaurenNguyen14**

### CODE REVIEW

File: common/main/java/com/couchbase/lite/internal/core/C4QueryObserver.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    public C4QueryObserver newObserver(C4Query c4Query) {
+        // TODO: implement newObserver()
+        return null;
+    }
+
+    public void setEnabled(boolean enabled) {
+        //TODO: implement setEnabled()
+    }
+
+    public void getEnumerator(boolean forget) throws CouchbaseLiteException {
```
Comment (LaurenNguyen14) [inline common/main/java/com/couchbase/lite/internal/core/C4QueryObserver.java:35]: This should return the enumerator

**Review by bmeike — APPROVED** [review]
You are going to the a static constructor that hands the C interface implementation to a new instance of C4QueryObserver.  The constructor should be private.  You might not be able to make it private, but if you can’t you need to flag it as @VisibleForTesting

You are going to need a finalizer in there, in case close doesn’t get called.

I think the next step should be writing some tests.

Please go ahead and commit this much.

File: common/main/java/com/couchbase/lite/internal/core/C4QueryObserver.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        long nCreate(long c4Query);
+        void nSetEnabled(long handle, boolean enabled);
+        void nFree(long handle);
+        long nGetEnumerator(long handle, boolean forget) throws LiteCoreException;
+    }
+
+
+    @VisibleForTesting
+    static NativeImpl nativeImpl = new NativeC4QueryObserver();
+
```
Comment (bmeike) [inline common/main/java/com/couchbase/lite/internal/core/C4QueryObserver.java:23]: I think you still need the static constructor...

**Review by bmeike — APPROVED** [review]
Still need the static constructor.  Looks really good, tho

**Pull request merged**